File: backend/app/schemas/task.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field

from app.schemas import CategoryResponse


class TaskBase(BaseModel):
    body: str = Field(..., max_length=255)


class TaskCreate(TaskBase):
    category_id: UUID = Field(..., alias="categoryId")


class TaskUpdate(TaskBase):
    body: str | None = Field(default=None, max_length=255)
    completed: bool | None = Field(default=None)
    category_id: UUID | None = Field(default=None, alias="categoryId")


class TaskResponse(BaseModel):
    id: UUID
    body: str
    completed: bool
    category: CategoryResponse
    created_at: datetime = Field(..., alias="createdAt")
    updated_at: datetime = Field(..., alias="updatedAt")

    class Config:
        orm_mode = True

File: frontend/src/interfaces/task.ts
import { Category } from './category';

export interface Task {
  id: string;
  body: string;
  completed: boolean;
  categoryId: string;
  createdAt: string;
  updatedAt: string;
}

export interface TaskWithCategory extends Omit<Task, 'categoryId'> {
  category: Category;
}
export type TaskCreate = Pick<Task, 'body' | 'categoryId'>;
export type TaskUpdate = Partial<Pick<Task, 'body' | 'completed' | 'categoryId'>>;
export type TaskMarkAsCompleted = Pick<Task, 'completed'>;
export type TaskId = Pick<Task, 'id'>;
